Migrate from Travis to GitHub Actions

crankyoldgit opened the pull request:
* Move any remaining Travis processes to GitHub Actions
* Remove Travis files and references. (Excluding `ReleaseNotes.md`)
* Change badges as well.
  - Remove two defunct badges.
  - Add a badge per GitHub Actions workflow.

Fixes #1522

crankyoldgit committed:
Don't run unit tests under Travis.
These are being run as GitHub actions. Reduce stuff running on Travis

For #1522

crankyoldgit committed:
Attempt to build some examples using Github Actions & PlatformIO

crankyoldgit commented:
@NiKiZe Hold ya horses! I hadn't added you as a reviewer yet. I'm still working on this. ;-)
In order for the Github Actions etc to trigger/run I had to create a PR.
As I work to clean this up/improve etc, I'll be adding more and more. Some will work. Some will fail.

I'll let you know when it's ready to review! ;-)

NiKiZe commented:
Is there any way to cache platformio compiles to reduce time spent? Seeing the output the same files are compiled multiple times?

crankyoldgit commented:
> Is there any way to cache platformio compiles to reduce time spent? Seeing the output the same files are compiled multiple times?

Yes, and it is set up to do that. If the compile/build options are the same, it does cache them.

However, a lot of the "duplication" is due to different compiler flags being used. e.g. A build with `-D_IR_LOCALE_=en-AU`, then a build with `-D_IR_LOCALE_=de-DE`. When that happens, it can't cache the files. :-(

crankyoldgit commented:
> > Is there any way to cache platformio compiles to reduce time spent? Seeing the output the same files are compiled multiple times?
> 
> Yes, and it is set up to do that. If the compile/build options are the same, it does cache them.
> 
> However, a lot of the "duplication" is due to different compiler flags being used. e.g. A build with `-D_IR_LOCALE_=en-AU`, then a build with `-D_IR_LOCALE_=de-DE`. When that happens, it can't cache the files. :-(

Well, let's just say, that's what it _was_ doing under Travis, I haven't got this Github Actions stuff sorted yet.

crankyoldgit committed:
More workflows & changes
* Make a workflow for linters
* PlatformIO workflow now does all the examples.
  - Try to improve caching.
* Try to improve UnitTest speed by "-j 2" for Make.
* Remove the moved stuff from Travis

For #1526

crankyoldgit committed:
Fix some python linter errors

crankyoldgit committed:
Another attempt to fix linter error

crankyoldgit committed:
Try to fix cpplint

crankyoldgit committed:
Move the Misc checks to GitHub Actions.
* Remove moved checks from Travis
* Cleanup & improvement to Library Linter workflow.

crankyoldgit committed:
Fix errors library manager workflow
* Also more cleanup.

crankyoldgit committed:
Another fix.

crankyoldgit committed:
Fix error in name

crankyoldgit committed:
Move doxygen check to workflow
* Doxygen check moved to github workflow.
* No more travis checks, so remove file.

crankyoldgit committed:
Update status badges.
* Move from Travis to the new GitHub Workflows.
* Remove old non-updating badges.

For #1522

crankyoldgit committed:
Fix badge issue
* Cut & paste fail. Fixed.
* Add to FR and DE files.

For #1522

crankyoldgit committed:
Use better names in the Workflows
Turns out the status badges use these to name the badges. So pick ones we'd like to see in the badges.

For #1522

crankyoldgit committed:
Make the new status badges be links to the relevent actions.

crankyoldgit committed:
Hopefully with the correct URL this time.

NiKiZe commented:
There is no risk of running out of free credits for these actions?

crankyoldgit commented:
> There is no risk of running out of free credits for these actions?

They don't seem to have a quota system, for the free/public tier. So ... dunno.

crankyoldgit committed:
Rename workflow files & remove Travis references.
* Make the workflow `.yml` file names more consistent.
* Remove references to Travis.

crankyoldgit commented:
Okay @NiKiZe this PR is done I think. Feel free to review it. ;-)

crankyoldgit commented:
> There is no risk of running out of free credits for these actions?

Just double checked. Scroll down a fair bit on https://github.com/features/actions. It's "Free" for Open Source. ;-) \<phew>

NiKiZe commented on README.md:
```diff
@@ -1,9 +1,10 @@
+[![Tests](https://github.com/crankyoldgit/IRremoteESP8266/actions/workflows/UnitTests.yml/badge.svg)](../../actions/workflows/UnitTests.yml)
+[![Documentation](https://github.com/crankyoldgit/IRremoteESP8266/actions/workflows/Documentation.yml/badge.svg)](../../actions/workflows/Documentation.yml/badge.svg)
 [![arduino-library-badge](https://www.ardu-badge.com/badge/IRremoteESP8266.svg?)](https://www.ardu-badge.com/IRremoteESP8266)
-[![Average time to resolve an issue](http://isitmaintained.com/badge/resolution/crankyoldgit/IRremoteESP8266.svg)](http://isitmaintained.com/project/crankyoldgit/IRremoteESP8266 "Average time to resolve an issue")
-[![Percentage of issues still open](http://isitmaintained.com/badge/open/crankyoldgit/IRremoteESP8266.svg)](http://isitmaintained.com/project/crankyoldgit/IRremoteESP8266 "Percentage of issues still open")
```
For reference, these are no longer updating, which was clarified in commit message, but had me confused at first.

NiKiZe commented on .github/workflows/UnitTests.yml:
```diff
@@ -16,6 +16,10 @@ jobs:
     - name: Install the Google test suite
       run: (cd test; make install-googletest)
```
should -j 2 was here as well?

crankyoldgit replied:
Good catch. I'll fix it.

crankyoldgit replied:
Done.

NiKiZe reviewed (approved):
Awesome!

NiKiZe commented:
> > There is no risk of running out of free credits for these actions?
> 
> Just double checked. Scroll down a fair bit on https://github.com/features/actions. It's "Free" for Open Source. ;-) <phew>

It only mentions private repos (to my eyes) ... the Free part was
# to big

NiKiZe commented:
Any chance we can limit number of languages built? Or maybe build everything in order of language instead?

crankyoldgit commented:
> Any chance we can limit number of languages built? Or maybe build everything in order of language instead?

Agreed. We need to do something better.
I'd prefer if they were only run/compiled when the `locale` stuff changed our for a release.

I don't know how to enact that yet.

crankyoldgit committed:
Add "-j 2" for google test suite
- Caught in code review.
